src/pages/api/rss.ts

```diff
@@ -0,0 +1,107 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { fetchWithTimeout } from 'utils/fetch-utils';
+import Cache from 'node-cache';
+import RSS from 'rss';
+
+interface feedItem {
+    title: string;
+    url: string;
+    pubDate?: string;
+    description?: string;
+}
+
+// Misc config settings
+// ----------------------------------------
+const millisecondsToFetchTimeout = 60 * 1000; // 60 seconds timeout in ms.
+const secondsToCacheExpiration = 60 * 10; // 10 minutes expiration.
+const rssUrl = `${process.env.XP_ORIGIN}/_/service/no.nav.navno/rss`;
+const rssServiceSecret = process.env.SERVICE_SECRET;
+const cacheKey = 'rss-cache';
+
+const cache = new Cache({
+    stdTTL: secondsToCacheExpiration,
+    deleteOnExpire: false, // We still want to serve expired cache while a new fetch is being fetched in background.
+});
+
+// TODO: Util function to use from rss and sitemap
+const processResponse = async (response: Response) => {
+    if (response.ok) {
+        const rssFeed: feedItem[] = await response.json();
+        return rssFeed;
+    } else {
+        console.error(
+            `Error while fetching RSS data: ${response.status} - ${response.statusText}`
+        );
+        return null;
+    }
+};
+
+const saveToCache = (xml: any): void => {
+    cache.set(cacheKey, xml);
+};
+
+const fetchRSSFeedAndUpdateCache = async (url: string) => {
+    const fetchOptions: RequestInit = {
+        headers: { secret: rssServiceSecret },
+    };
+    const response = await fetchWithTimeout(
+        url,
+        millisecondsToFetchTimeout,
+        fetchOptions
+    );
+    const jsonFeed = await processResponse(response);
+    if (!jsonFeed) {
+        return null;
+    }
+    const rssFeed = new RSS({
+        title: 'Nyheter fra NAV',
+        description: 'www.nav.no',
+        feed_url: 'https://www.nav.no/no/rss',
```

Comment:
Skal denne være nav.no/rss? Jfr next.config

Reply:
Foreslår at vi beholder dagens url som feed-url, men åpner for www.mnav.no/rss i tillegg.